### DISCUSSION TITLE
Datetime formats

### DISCUSSION THREAD

**Discussion by Mastomaki** [Spine-project/Spine-Toolbox#2832] [Support]
Hello!

Can you provide a list of datetime formats which the importer accepts?

**Comment by soininen** [comment]
The recommended format is [ISO8601](https://fi.wikipedia.org/wiki/ISO_8601) which is fast. If non-ISO8601 format is detected, Importer falls back to the `dateutil` Python library which supports *a lot* of different formats and even vaguely defined datetimes. Perhaps best to check their [documentation](https://dateutil.readthedocs.io/en/stable/parser.html) for more information although it does not contain an exhaustive list of supported formats.

**Comment by Mastomaki** [comment]
I could add this to documentation if someone tells how.

**Reply by soininen** [reply]
That would be very nice contribution!

Basically, you would start by opening an issue in our [issue tracker](https://github.com/spine-tools/Spine-Toolbox/issues) and assign yourself to it. Then, you would clone the Spine-Toolbox repository (if you have not already done so), create a new Git branch and make the changes (in `<repository root>/docs/source/data_import_export.rst` I guess). To build the documentation yourself, you need to first install the developer dependencies with `python -mpip install -r dev-requirements.txt`. Then, run the script `bin\build_doc` (Windows) or `bin/build_doc.py` (*NIX). After the script finishes, you can open `<repository root>/docs/build/html/index.html` in browser to check that everything looks good.

When you are happy with the changes, you need to commit your changes to the Git repository, push the commit to GitHub and open a [Pull request](https://github.com/spine-tools/Spine-Toolbox/pulls). Somebody from the developers will review the changes and merge them to the master branch once all kinks have been ironed out.

Let me know if you need more detailed instructions with any of the steps.

*Edit: fixed script names*

**Comment by Mastomaki** [comment]
Here is the error I get. I am not sure about downgrading because it would probably bring errors elsewhere.

```
Spine-Toolbox/bin$ python build_ui.py 
<Script for Building Spine Toolbox GUI>
Copyright (C) <2017-2021>  <Spine project consortium>
This program comes with ABSOLUTELY NO WARRANTY; for details see 'about'
box in the application. This is free software, and you are welcome to
redistribute it under certain conditions; See files COPYING and
COPYING.LESSER for details.
--- Entering /home/jube/data/spine/versions/v08/Spine-Toolbox/spinetoolbox/spine_db_editor/ui ---
--- Entering /home/jube/data/spine/versions/v08/Spine-Toolbox/spinetoolbox/ui ---
Building settings.py

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/home/jube/anaconda3/envs/spinetb08/bin/pyside6-uic", line 5, in <module>
    from PySide6.scripts.pyside_tool import uic
  File "/home/jube/anaconda3/envs/spinetb08/lib/python3.11/site-packages/PySide6/__init__.py", line 124, in <module>
    _setupQtDirectories()
  File "/home/jube/anaconda3/envs/spinetb08/lib/python3.11/site-packages/PySide6/__init__.py", line 64, in _setupQtDirectories
    from shiboken6 import Shiboken
  File "/home/jube/anaconda3/envs/spinetb08/lib/python3.11/site-packages/shiboken6/__init__.py", line 27, in <module>
    from shiboken6.Shiboken import *
Appending license to settings.py
settings.ui seems to have a license already. Skipping.
--- Build completed ---
```

**Reply by soininen** [reply]
I just realized I mentioned the wrong script above! You should run `bin\build_doc` (`bin/build_doc.py` on *NIX) to build the documentation, not `bin\build_ui` which builds the Toolbox UI files. I am very sorry for the mistake!

I will need to have a look at that error from `numpy`, though. We have fixed some issues with Numpy 2 recently but maybe there is more it.

**Reply by Mastomaki** [reply]
Ok now it works!
